### PR TITLE
Add missing chrome debugging apis for Sync.User

### LINKED ISSUE #791: Add missing API's for chrome debugging
A number of API's, especially related to sync, are missing from the chrome debugging proxies.

### PR DESCRIPTION
**Pull request by kristiandupont**
This resolves #791.

### CODE REVIEW

File: src/rpc.cpp [hunk 1]
```diff
@@ -133,6 +134,27 @@ RPCServer::RPCServer() {
         RPCObjectID realm_id = store_object(realm_object);
         return (json){{"result", realm_id}};
     };
+    m_requests["/create_user"] = [this](const json dict) {
```
Comment (alazier) [inline src/rpc.cpp]: Why not just forward these requests to the Sync object as we do with all other methods? Same question for the rest of new sync methods.

Comment (kristiandupont) [inline src/rpc.cpp]: Maybe I don't understand the implementation well enough, but if we create the user simply with a `/call_method` request, the object wont be stored in `m_objects`, so we would not be able to access it afterwards? Or are you suggesting that I add that ability to `call_method`?

Comment (alazier) [inline src/rpc.cpp]: Objects are stored when they are serialized before they are returned. Not sure why this wouldn't just work - although I haven't looked at this code for quite a while.

Comment (alazier) [inline src/rpc.cpp]: Looks like you are already calling `store_object` in `serialize_json_value`, so the object should be stored.

Comment (kristiandupont) [inline src/rpc.cpp]: Yes, I think you are right. I am trying out just sending the `Sync.User` object in `create_session` and use it like the other objects.

File: tests/js/index.js [hunk 2]
```diff
@@ -28,8 +28,8 @@ var TESTS = {
     QueryTests: require('./query-tests'),
     EncryptionTests: require('./encryption-tests'),
     MigrationTests: require('./migration-tests'),
-    UserTests: require('./user-tests'),
-    AsyncTests: require('./async-tests')
+    UserTests: require('./user-tests')
+    /*AsyncTests: require('./async-tests')*/
```
Comment (alazier) [inline tests/js/index.js]: why are we disabling these? isn't this the only place we actually test collection notifications?

**Pull request merged**